### ISSUE TITLE
Add possibility to load >=3 dimensional data from CDF files

### ISSUE THREAD

**Issue by jgieseler** [sunpy/sunpy#6194]
### Describe the feature

At the moment, `read_cdf()` skips data columns that are >=3 dimensional data (cf. https://github.com/sunpy/sunpy/issues/5869 & https://github.com/sunpy/sunpy/pull/5975). I was wondering if there are some possibilities to load this data without too much effort because otherwise we're missing out some nice new spacecraft data.

### Proposed solution

The crucial part is at https://github.com/sunpy/sunpy/blob/main/sunpy/io/cdf.py#L92= where a column is skipped if its dimension is > 2. Would it be a possibility to rewrite it to reflect the dependencies in the column name, like it's done 3 lines below for dim = 2 data:

``` python
elif data.ndim == 2:
    # Multiple columns, give each column a unique label
    for i, col in enumerate(data.T):
        df[var_key + f'_{i}'] = col
        units[var_key + f'_{i}'] = unit
```

So for example, for the `var_key` ELECTRON_FLUX that has (next to EPOCH) an ENERGY and VIEWING dependency, something like "Electron_Flux_Energy1_Viewing1", "Electron_Flux_Energy1_Viewing2", "Electron_Flux_Energy2_Viewing1", and so on. Then we still have a standard DataFrame that is passed to GenericTimeSeries in [L105](https://github.com/sunpy/sunpy/blob/main/sunpy/io/cdf.py#L105=)

**Comment by AthKouloumvakos** [comment]
Yes, multi-dim data (>=2) may need something more than 'TimeSeries'.

 'xarray' seems a nice option. I am working on this for the last few weeks and seems very good.
 
 So, inspired by https://github.com/jgieseler/solo-epd-loader I am making some tests for xarray and SOLO-HET data.
 
I have constructed an xarray.Dataset() successfully for the dataset. There are many advantages using xarray as I can tell so far. The data multi-dim access is very easy, you can concat data also easy, add a new dimention easy, resample easy, ploting at any dim easy. One clear advanage is that you have flux, uncertainty, rate in the same dataset, and time and energy are the two axis of the dataset, also I like the attrs option (see bellow).
 
An example:

```
import cdflib
import sunpy_soar
import sunpy.net.attrs as a
import xarray as xr
from cdflib.epochs import CDFepoch
from sunpy.net import Fido
from sunpy.timeseries import TimeSeries


# Create search attributes
instrument = a.Instrument("EPD")
time = a.Time("2021-02-01", "2021-02-02")
level = a.Level(2)

product = a.soar.Product("EPD-HET-ASUN-RATES")

# Do search
result = Fido.search(instrument & time & level & product)
print(result)

# Download files
files = Fido.fetch(result)
print(files)

parameters = cdflib.CDF(files[0])
index = CDFepoch.to_datetime(parameters.varget('EPOCH'))
data_ = {}
data_['H'] = xr.Dataset(
    {
    'flux': (['time', 'energy'], parameters.varget('H_Flux')),
    'uncertainty': (['time', 'energy'], parameters.varget('H_Uncertainty')),
    'rate': (['time', 'energy'], parameters.varget('H_Rate'))
    },
    coords={'time':index, 'energy':parameters.varget('H_Bins_Low_Energy')},
    attrs={'units':{'time':'UTC', 'energy':'MeV', 'flux':'#/(cm^2*s*sr*Mev/nuc.)'}
         })
```

`print(data_['H'])` gives:

```
<xarray.Dataset>
Dimensions:      (time: 17280, energy: 36)
Coordinates:
  * time         (time) datetime64[ns] 2021-02-01T00:00:00.946408 ... 2021-02...
  * energy       (energy) float32 7.045 7.354 7.89 8.484 ... 81.64 89.46 97.63
Data variables:
    flux         (time, energy) float32 0.0 0.0 0.0 0.0 0.0 ... 0.0 0.0 0.0 0.0
    uncertainty  (time, energy) float32 0.0 0.0 0.0 0.0 0.0 ... 0.0 0.0 0.0 0.0
    rate         (time, energy) float32 0.0 0.0 0.0 0.0 0.0 ... 0.0 0.0 0.0 0.0
Attributes:
    units:    {'time': 'UTC', 'energy': 'MeV', 'flux': '#/(cm^2*s*sr*Mev/nuc.)'}
```

**Comment by jgieseler** [comment]
Well, I was in the process of writing a long text laying out my motivation, but after cross-checking with the [PSP/ISOIS User Guide](https://spdf.gsfc.nasa.gov/pub/data/psp/isois/ISOIS_Data_Glossary.pdf) I'm now also hesitating to continue my suggested "easy solution". Because in the past these in-situ particle measurements were what I would call 'pseudo' multidimensional data, where usually only `time` was a really 'free' dimension, whereas `energy` and `viewing` usually were confined to defined bins that don't change at all (like in the example @AthKouloumvakos showed for SolO/HET). But I think for PSP this really changed, and introducing an "easy solution" now might actually cause more problems than helping because people won't cross-check how the dependencies really are.

So in conclusion, I agree that we should either introduce a really thorough approach, or leave it as it's now.

@AthKouloumvakos, you should take a look at the PSP datasets I mentioned in https://github.com/sunpy/sunpy/issues/5869 because these have more dependencies than SolO/HET. But let's talk about that in person (virtually 😉).
